### PR TITLE
[IMP] menu: allow to add a secondary icon to a menu

### PR DESCRIPTION
**Pull request by hokolomopo**
## [IMP] menu: allow to add a secondary icon to a menu:

This commit allows to add a secondary icon to a menu. This icon is
displayed on the right of the menu item.

Also take the occasion to bootstrap-ify the menu component.

## [IMP] headers overlay: simplify CSS & use bootstrap

This commit simplifies the CSS of the headers overlay and uses
bootstrap classes instead of custom ones.

This also improves the "ARROW-*" icons by centering the actual icon
inside the SVGs.

Task: : [3487684](https://www.odoo.com/web#id=3487684&action=333&active_id=2328&model=project.task&view_type=form&cids=1&menu_id=4720)

## review checklist

- [ ] feature is organized in plugin, or UI components
- [ ] support of duplicate sheet (deep copy)
- [ ] in model/core: ranges are Range object, and can be adapted (adaptRanges)
- [ ] in model/UI: ranges are strings (to show the user)
- [ ] undo-able commands (uses this.history.update)
- [ ] multiuser-able commands (has inverse commands and transformations where needed)
- [ ] new/updated/removed commands are documented
- [ ] exportable in excel
- [ ] translations (\_t("qmsdf %s", abc))
- [ ] unit tested
- [ ] clean commented code
- [ ] track breaking changes
- [ ] doc is rebuild (npm run doc)
- [ ] status is correct in Odoo

### CODE REVIEW

File: src/components/headers_overlay/headers_overlay.ts [hunk 1]
```diff
@@ -486,8 +477,8 @@ export class ColResizer extends AbstractResizer {
     });
   }
 
-  unhideStyleValue(hiddenIndex: HeaderIndex): Pixel {
-    return this._getDimensionsInViewport(hiddenIndex).start;
+  getHideButtonsStyle(hiddenIndex: HeaderIndex): string {
```
Comment (anhe-odoo) [inline src/components/headers_overlay/headers_overlay.ts:480]: getUnhideButtonsStyle ?

File: src/components/headers_overlay/headers_overlay.ts [hunk 2]
```diff
@@ -687,8 +669,8 @@ export class RowResizer extends AbstractResizer {
...
```
Comment (anhe-odoo) [inline src/components/headers_overlay/headers_overlay.ts:672]: same comment here

File: src/components/icons/icons.xml [hunk 3]
```diff
@@ -303,23 +303,23 @@
...
-    <svg class="o-icon">
-      <polygon fill="currentColor" points="0 0 4 4 0 8" transform="translate(5 5)"/>
+    <svg class="o-icon" transform="rotate(-90)">
+      <t t-call="o-spreadsheet-Icon.TRIANGLE_DOWN"/>
     </svg>
   </t>
   <t t-name="o-spreadsheet-Icon.TRIANGLE_LEFT">
-    <svg class="o-icon">
-      <polygon fill="currentColor" points="4 0 0 4 4 8" transform="translate(5 5)"/>
+    <svg class="o-icon" transform="rotate(90)" viewBox="0 0 18 18">
```
Comment (anhe-odoo) [inline src/components/icons/icons.xml:321]: (didn't try by myself), do you still need the viewBox for this one ?

File: src/components/menu/menu.xml [hunk 4]
```diff
@@ -18,7 +18,7 @@
               t-att-data-name="menuItem.id"
               t-on-click="(ev) => this.onClickMenu(menuItem, menuItem_index, ev)"
               t-on-mouseover="(ev) => this.onMouseOver(menuItem, menuItem_index, ev)"
-              class="o-menu-item"
+              class="o-menu-item d-flex align-items-center"
```
Comment (anhe-odoo) [inline src/components/menu/menu.xml:21]: The justify-content: space-between is not present anymore. Is it intended ?

Comment (hokolomopo) [reply src/components/menu/menu.xml:21]: Yeah it was useless, it's kinda overwritten by the `ms-auto` of the children

**Review by LucasLefevre — APPROVED** [review]
robodoo r+ rebase-ff

**Comment by LucasLefevre** [comment]
robodoo r+ rebase-ff

**Pull request closed without merging**